src/modules/flow/http-client/http-client.c

```diff
@@ -402,17 +371,102 @@ common_request_finished(void *data,
         "%s Could not parse url contents ", mdata->url);
 }
 
+
+static ssize_t
+sse_received_data_cb(void *data, const struct sol_http_client_connection *conn,
+    struct sol_buffer *buf)
+{
+    struct sol_flow_node *node = data;
+    const size_t prefix_len = strlen("data: ");
+    const size_t suffix_len = strlen("\n\n");
+    const struct http_client_node_type *type;
+    struct sol_str_slice slice;
+    size_t consumed = 0;
+
+    SOL_DBG("Received SSE Data - *%.*s*",
+        SOL_STR_SLICE_PRINT(sol_buffer_get_slice(buf)));
+
+    slice = sol_buffer_get_slice(buf);
+    if (!memmem(slice.data, slice.len, "\n\n", suffix_len))
+        return 0;
+
+    type = (const struct http_client_node_type *)sol_flow_node_get_type(node);
+
+    while (slice.len) {
+        char *start, *end, *content;
+        struct sol_buffer content_buf;
+        size_t content_len, total_len;
+        int r;
+
+        start = memmem(slice.data, slice.len, "data: ", prefix_len);
+        SOL_NULL_CHECK(start, -EINVAL);
+
+        end = memmem(slice.data, slice.len, "\n\n", suffix_len);
+        if (!end) //Wait for more data
+            goto exit;
+
+        content = start + prefix_len;
+        content_len = end - content;
+
+        sol_buffer_init_flags(&content_buf, content, content_len,
+            SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED);
+        content_buf.used = content_len;
+        total_len = content_len + prefix_len + suffix_len;
+        consumed += total_len;
+
+        SOL_DBG("Parsed SSE data:*%.*s*",
+            SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&content_buf)));
+
+        if (type->process_token)
+            r = type->process_token(node, &content_buf);
+        else
+            r = type->process_data(node, &content_buf);
+        sol_buffer_fini(&content_buf);
+        SOL_INT_CHECK(r, < 0, r);
+        slice.len -= total_len;
+        slice.data += total_len;
+    }
+
+exit:
+    SOL_DBG("Buf len: %zu - Consumed: %zu", buf->used, consumed);
+    return consumed;
```

Comment:
if we don't consume the data (ie; partial "data: ..." without \n\n), then the infrastructure will keep that remaining bytes for the next call?